src/components/Button/src/Button.vue

```diff
@@ -130,37 +144,110 @@ export default {
 			default: undefined,
 		},
 		/**
-		 * Background color of button
+		 * Main color of button
 		 */
 		color: {
 			type: String,
 			default: undefined,
 			validator: (color) => colord(color).isValid(),
 		},
 		/**
-		 * Text color of button
+		 * Main hover color of button
+		 */
+		colorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
+		/**
+		 * Text color of button (only applied on fill buttons)
 		 */
 		textColor: {
 			type: String,
 			default: undefined,
 			validator: (color) => colord(color).isValid(),
 		},
+		/**
+		 * Text hover color of button (only applied on fill buttons)
+		 */
+		textColorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
 		/**
 		 * Variant
-		 * @values primary, secondary, tertiary
+		 * @values fill, outline, ghost
 		 */
 		variant: {
 			type: String,
 			default: undefined,
-			validator: (variant) => ['primary', 'secondary', 'tertiary'].includes(variant),
+			validator: (variant) => ['fill', 'outline', 'ghost'].includes(variant),
 		},
 		/**
 		 * Shape of button
+		 * @values squared, rounded, pill, 'Npx', 'N%'
 		 */
 		shape: {
 			type: String,
 			default: undefined,
-			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape),
+			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape) || shape.match(/\d*\.?\d+(?:px|%)/i),
+		},
+		/**
+		 * Shape of button
+		 * @values 'Npx', 'N%'
+		 */
+		shapeHover: {
+			type: String,
+			default: undefined,
+			validator: (shape) => shape.match(/\d*\.?\d+(?:px|%)/i),
+		},
+		/**
+		 * Border width of button (e.g. '3px')
+		 */
+		borderWidth: {
+			type: String,
+			default: undefined,
+		},
+		/**
+		 * Border hover width of button (e.g. '3px')
+		 */
+		borderWidthHover: {
+			type: String,
+			default: undefined,
+		},
+
+		/**
+		 * Border color of button (only applied on fill buttons)
+		 */
+		borderColor: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
+		/**
+		 * Border hover color of button (only applied on fill buttons)
+		 */
+		borderColorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
+		/**
+		 * Box-shadow of button (e.g. 0px 0px 1px 3px #000000 or 0px 0px 1px 3px rgba(0, 0, 0, 0.5))
+		 */
+		boxShadow: {
+			type: String,
+			default: undefined,
+			validator: (boxShadow) => boxShadow.match(/(\d*\.?\d+px\s+){4}(#[\da-z]{6}|rgba\((\s*\d+\s*,){3}\s*[\d.]+\))/i),
```

Comment:
`cssValidator('box-shadow')` or if the regex was explicitly chosen because it limits some undesirable values that `cssValidator('box-shadow')` would otherwise allow then we should at least refactor this regex into a constant above, e.g.
```
// validates CSS box-shadow values
const BOX_SHADOW_REGEX = /(\d*\.?\d+px\s+){4}(#[\da-z]{6}|rgba\((\s*\d+\s*,){3}\s*[\d.]+\))/i;
```